### PR TITLE
Add joint denial (NOR)

### PR DESCRIPTION
**Pull request by remimimimimi**
Implementation of #2818. 

What should I do with symbol representation? I was not able to find free symbol for it.

### CODE REVIEW

**Comment by tirix** [comment]
Looks good, very nice first submission!

I think the symbol would be `&barvee;`, a vee with an overbar: ⊽
That's `&#8893;` in HTML encoding.
The LaTeX symbol would be `\barvee`, [but I'm not sure if we have it](https://tex.stackexchange.com/questions/201027/in-math-mode-i-have-access-to-veebar-but-not-barvee-why).

**Comment by remimimimimi** [comment]
Thanks!

The current included latex packages are amssymb, amsmath, mathtools, mathrsfs, phonetic, and graphicx. However, we don't have unicode-latex which contains `\barvee`.

I think that adding a whole new latex package deserves its own discussion, and for now `\bar{\vee}` will be fine.

If we want to have proper displaying (coherent with `\veebar`) we can add just that symbol manually, but that seems quite dirty.

**Comment by tirix** [comment]
Yes this looks good for now!

I'll let @benjub comment about whether we take this submission directly into the "main" part of set.mm or if he prefers a submission into a mathbox first, since he was the one who initially suggested adding NOR.

File: set.mm [hunk 1]
```diff
@@ -431454,6 +431526,12 @@ orthogonal vectors (i.e. whose inner product is 0) is the sum of the
   althtmldef "\/_" as " &#8891; ";
     /* 2-Jan-2016 reverted sans-serif */
   latexdef "\/_" as "\veebar";
+htmldef "-\/" as
+    " <IMG SRC='vee.gif' WIDTH=9 HEIGHT=19 ALT=' -\/' TITLE='-\/' " +
+    " STYLE='text-decoration:overline;'> ";
+  althtmldef "-\/" as
+    ' <span>&#8893;</span> ';
+  latexdef "-\/" as "\bar{\vee}";
```
Comment (benjub) [inline set.mm:431534]:
```suggestion
  latexdef "-\/" as "\mathrel{\overline{\vee}}";
```

File: set.mm [hunk 2]
```diff
@@ -431454,6 +431526,12 @@ orthogonal vectors (i.e. whose inner product is 0) is the sum of the
   althtmldef "\/_" as " &#8891; ";
     /* 2-Jan-2016 reverted sans-serif */
   latexdef "\/_" as "\veebar";
```
Comment (benjub) [inline set.mm:431528]:
```suggestion
  latexdef "\/_" as "\mathrel{\underline{\vee}}";
```

File: set.mm [hunk 3]
```diff
@@ -11655,6 +11656,52 @@ This definition (in the form of ~ dfifp2 ) appears in Section II.24 of
...
+  $( Define joint denial ("not-or" or "nor").  Represented by a vertical bar,
+     but we use a different symbol to avoid ambiguity with other uses of the
+     vertical bar.  After we define the constant true ` T. ` ( ~ df-tru ) and
+     the constant false ` F. ` ( ~ df-fal ), we will be able to prove these
+     truth table values: ` ( ( T. -\/ T. ) <-> F. ) ` ( ~ trunortru ),
+     ` ( ( T. -\/ F. ) <-> F. ) ` ( ~ trunorfal ), ` ( ( F. -\/ T. ) <-> F. ) `
+     ( ~ falnortru ), and ` ( ( F. -\/ F. ) <-> T. ) ` ( ~ falnorfal ).
+     Contrast with ` /\ ` ( ~ df-an ), ` \/ ` ( ~ df-or ), ` -> ` ( ~ wi ),
+     ` -/\ ` ( ~ df-nan ), and ` \/_ ` ( ~ df-xor ).  (Contributed by Remi,
+     25-Nov-2023.) $)
```
Comment (avekens) [inline set.mm:11680]: This is in the future ;-)

File: set.mm [hunk 4]
```diff
@@ -11655,6 +11656,52 @@ This definition (in the form of ~ dfifp2 ) appears in Section II.24 of
...
+  df-nor $a |- ( ( ph -\/ ps ) <-> -. ( ph \/ ps ) ) $.
+
+  $( The connector ` -\/ ` is commutative.  (Contributed by Remi,
+     25-Oct-2023.) $)
+  norcom $p |- ( ( ph -\/ ps ) <-> ( ps -\/ ph ) ) $=
+    ( wo wn wnor orcom notbii df-nor 3bitr4i ) ABCZDBACZDABEBAEJKABFGABHBAHI $.
+
+  $( ` -. ` is expressable via ` -\/ ` .  (Contributed by Remi,
```
Comment (avekens) [inline set.mm:11688]: "expressible"?

File: set.mm [hunk 5]
```diff
@@ -12167,6 +12214,31 @@ logical connectives we have introduced (either axiomatically or by a
     ( wfal wxo wtru wn wb df-xor falbifal xchbinx nottru bitri ) AABZCDAKAAECAA
     FGHIJ $.
 
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Joint denial
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  $( A ` -\/ ` identity.  (Contributed by Remi, 25-Oct-2023.) $)
```
Comment (avekens) [inline set.mm:12223]: Maybe we should be a little bit more verbose here (I know the other theorems of this kind have also these short comments). I would suggest:

``A ` -\/ ` identity: true nor true is false.``

File: set.mm [hunk 6]
```diff
@@ -11655,6 +11656,52 @@ This definition (in the form of ~ dfifp2 ) appears in Section II.24 of
...
+
+  $( Declare connective for joint denial ("nor"). $)
+  $c -\/ $.  $( Overlined "vee" (read:  "nor") $)
+
+  $( Extend wff definition to include joint denial ("nor"). $)
+  wnor $a wff ( ph -\/ ps ) $.
+
+  $( Define joint denial ("not-or" or "nor").  Represented by a vertical bar,
+     but we use a different symbol to avoid ambiguity with other uses of the
+     vertical bar.  After we define the constant true ` T. ` ( ~ df-tru ) and
```
Comment (benjub) [inline set.mm:11673]: That second sentence was there for df-nan, but should be removed for df-nor.

File: set.mm (hunk 5 above)
Comment (benjub) [reply set.mm:12223]: In general, I favor precise comments, like the one @avekens wrote, but in this particular case, I think the current comment is sufficient, similar to other comments in the section on truth tables.

**Review by benjub — APPROVED** [review]
Thanks, that's a nice submission.

Note to anyone: associativity is not proved yet !

**Review by benjub — CHANGES_REQUESTED** [review]
Sorry, I shouldn't have approved: there is my comment above about removing the superfluous sentence from a comment.

**Comment by remimimimimi** [comment]
> Note to anyone: associativity is not proved yet !

Because [it's a non-associative operation](https://proofwiki.org/wiki/NOR_is_not_Associative).

**Comment by benjub** [comment]
> > Note to anyone: associativity is not proved yet !
> 
> Because [it's a non-associative operation](https://proofwiki.org/wiki/NOR_is_not_Associative).

Ha, that's a fair reason !  It should be some result like ~nanass.

**Review by benjub — APPROVED** [review]

**Review by jkingdon — APPROVED** [review]
Part of me wonders if we should have a section header like "other connectives" to group together the propositional logic connectives other than the main five ones (negation, implication, or, and, and biconditional). Looks like such a section would have about six (or so) subsections. I know Norm was a bit worried about proliferating these things, and I guess from my point of view that's largely a question of organization, rather than whether to have them at all.

But I'm fine with that being food for thought, I don't see any need to delay merging this pull request.

**Comment by benjub** [comment]
> Part of me wonders if we should have a section header like "other connectives" to group together the propositional logic connectives other than the main five ones (negation, implication, or, and, and biconditional). Looks like such a section would have about six (or so) subsections. I know Norm was a bit worried about proliferating these things, and I guess from my point of view that's largely a question of organization, rather than whether to have them at all.

I think a dedicated section is not needed. As for the connectives, I think this MR added the final one ? Sure, technically, there is "does not imply", "is (not) implied by", and degenerated ones ((negated) projections, true/false as unary and binary operators). As for higher arity, I think ifp, had, cad are enough as well. (This is for classical logic.)

**Pull request merged**